src/transformers/models/mistral/modeling_mistral.py

```diff
@@ -680,7 +683,11 @@ def forward(
 
         # SDPA with memory-efficient backend is currently (torch==2.1.2) bugged with non-contiguous inputs with custom attn_mask,
         # Reference: https://github.com/pytorch/pytorch/issues/112577.
-        if query_states.device.type == "cuda" and attention_mask is not None:
+        if (
+            not _is_torch_version_greater_or_equal_than_2_2_0
+            and query_states.device.type == "cuda"
+            and attention_mask is not None
+        ):
             query_states = query_states.contiguous()
             key_states = key_states.contiguous()
             value_states = value_states.contiguous()
```

Comment:
If we are using torch2.2 this should not be super costly even if not necessary no?

Reply:
> If we are using torch2.2 this should not be super costly even if not necessary no?

Yes, you are right, it seems that not cost too much. I'll remove these three changes of making tensors non-contigous in torch 2.2.